crates/forge/src/lib.rs

```diff
@@ -285,146 +288,116 @@ fn try_close_tmp_dir(maybe_tmp_dir: Option<TempDir>) -> Result<()> {
-
-        fuzzing_happened |= was_fuzzed;
-
-        summaries.push(summary.clone());
-        if summary.runner_exit_status == RunnerStatus::TestFailed {
-            break;
-        }
+    let mut tasks = vec![];
+
+    for tests_from_crate in tests {
+        let tests_from_crate = Arc::new(tests_from_crate);
+        let runner_config = runner_config.clone();
+        let test_crate_type = tests_from_crate.test_crate_type;
+        let tests_len = tests_from_crate.test_cases.len();
+        tasks.push((
+            test_crate_type,
+            tests_len,
+            task::spawn({
+                async move { run_tests_from_crate(tests_from_crate, runner_config).await }
+            }),
+        ));
     }
+    for (test_crate_type, tests_len, task) in tasks {
+        pretty_printing::print_running_tests(test_crate_type, tests_len);
 
-    for tests_from_file in tests_iterator {
-        let skipped: Vec<TestCaseSummary> = tests_from_file
-            .test_cases
-            .iter()
-            .map(TestCaseSummary::skipped)
-            .collect();
-
-        for test_case_summary in &skipped {
-            pretty_printing::print_test_result(test_case_summary, None);
+        let (summary, was_fuzzed) = task.await??;
+        for test_case_summary in &summary.test_case_summaries {
+            pretty_printing::print_test_result(test_case_summary);
         }
-
-        let file_summary = TestCrateSummary {
-            test_case_summaries: skipped,
-            runner_exit_status: RunnerStatus::DidNotRun,
-            test_crate_type: tests_from_file.test_crate_type,
-        };
-        summaries.push(file_summary);
+        fuzzing_happened |= was_fuzzed;
+        summaries.push(summary.clone());
     }
 
     pretty_printing::print_test_summary(&summaries);
+
     if fuzzing_happened {
         pretty_printing::print_test_seed(runner_config.fuzzer_seed);
     }
 
     Ok(summaries)
 }
 
-fn run_tests_from_crate(
-    package_root: &Utf8PathBuf,
-    tests: TestsFromCrate,
-    runner_config: &RunnerConfig,
-    runner_params: &RunnerParams,
+async fn run_tests_from_crate(
+    tests: Arc<TestsFromCrate>,
+    runner_config: Arc<RunnerConfig>,
 ) -> Result<(TestCrateSummary, bool)> {
-    let runner = SierraCasmRunner::new(
-        tests.sierra_program,
-        Some(MetadataComputationConfig::default()),
-        OrderedHashMap::default(),
-    )
-    .context("Failed setting up runner.")?;
-
-    pretty_printing::print_running_tests(tests.test_crate_type, tests.test_cases.len());
+    let sierra_program = &tests.sierra_program;
+    let runner = Arc::new(
+        SierraCasmRunner::new(
+            sierra_program.clone(),
+            Some(MetadataComputationConfig::default()),
+            OrderedHashMap::default(),
+        )
+        .context("Failed setting up runner.")?,
+    );
 
     let mut was_fuzzed = false;
-    let mut results = vec![];
+    let mut tasks = FuturesOrdered::new();
```

Comment:
I wonder if we need this to be ordered.

Reply:
we need it in our e2e tests to have repetitivet output. That's the only one reason

Reply:
Okay, perhaps this should be a CLI configuration then? So you can enforce ordered execution if a flag (and we can use it tests) but the default should be unordered?

Reply:
@MaksymilianDemitraszek ?

Reply:
IMO in the first version we can leave `FuturesOrdered ` and we can change it in the next version

Reply:
I don't think it should be ordered, we should write e2e in such way that not requires exact order

Reply:
Looking from the user perspective I still think it should be ordered, the output will be really messy otherwise. We can create an issue for that and investigate `FuturesUnordered` makes us much faster - if it does ofc speed >>> UI

Reply:
+1, issue and investigation if <5% speedup is worth it

Reply:
I mean we don't have to do it here, but we should create an issue to investiagte it

Reply:
#843